.github/workflows/unit-and-integration-tests-unified.yml

```diff
@@ -41,6 +41,7 @@ on:
       - master
       - '[0-9]+.[0-9]+.[0-9]+' # release branches
       - '[0-9]+.[0-9]+.[0-9]+-[A-Za-z0-9]+' # release branches
+    types: [ opened, reopened, synchronize, labeled, unlabeled ]
```

Comment:
please don't add triggers on `labeled` / `unlabeled` ; this will implicitly cause further issues

Reply:
I am curious what kind of issues you are referring to? I am not super familiar with github actions.

Reply:
I have seen other projects using these events (calcite is one) - and was thinking about it some time ago....
I think the best is to separate jobs which are triggered on label changes and CI checks should be independent.

With the current setup: if someone adds an irrelevant label to PR it will right away abort all CI jobs - and start new ones with the `labeled` event (for which in the current setup it won't run unit tests).

A different approach could be to create a new job which is watching the labels and if a relevant label is changed it somehow restarts the main CI job